src/rpcrawtransaction.cpp

```diff
@@ -24,111 +23,280 @@ extern std::string GetTxProject(uint256 hash, int& out_blocknumber, int& out_blo
 extern void Imker(void *kippel);
 extern Upgrader upgrader;
 
+extern std::vector<std::pair<std::string, std::string>> GetTxStakeBoincHashInfo(const CMerkleTx& mtx);
+extern std::vector<std::pair<std::string, std::string>> GetTxNormalBoincHashInfo(const CMerkleTx& mtx);
+std::string TimestampToHRDate(double dtm);
+std::string GetPollXMLElementByPollTitle(std::string pollname, std::string XMLElement1, std::string XMLElement2);
+std::string GetShareType(double dShareType);
+
 #ifdef QT_GUI
 #include "qt/upgradedialog.h"
 extern Checker checker;
 #endif
 
-void GetTxStakeBoincHashInfo(json_spirit::mObject& res, const CMerkleTx& mtx)
+std::vector<std::pair<std::string, std::string>> GetTxStakeBoincHashInfo(const CMerkleTx& mtx)
 {
     assert(mtx.IsCoinStake() || mtx.IsCoinBase());
-
-    res["blockhash"]=mtx.hashBlock.GetHex();
+    std::vector<std::pair<std::string, std::string>> res;
 
     // Fetch BlockIndex for tx block
     CBlockIndex* pindex = NULL;
     CBlock block;
     {
         map<uint256, CBlockIndex*>::iterator mi = mapBlockIndex.find(mtx.hashBlock);
+
         if (mi == mapBlockIndex.end())
         {
-            res["error"] = "block_not_in_index";
-            return;
+            res.push_back(std::make_pair("ERROR", "Block not in index"));
+            return res;
         }
+
         pindex = (*mi).second;
+
         if (!block.ReadFromDisk(pindex))
         {
-            res["error"] = "block_read_failed";
-            return;
+            res.push_back(std::make_pair("ERROR", "Block read failed"));
+            return res;
         }
     }
 
     //Deserialize
     MiningCPID bb = DeserializeBoincBlock(block.vtx[0].hashBoinc,block.nVersion);
 
-    res["height"]=pindex->nHeight;
-    res["version"]=block.nVersion;
+    res.push_back(std::make_pair("Height", ToString(pindex->nHeight)));
+    res.push_back(std::make_pair("Block Version", ToString(block.nVersion)));
+    res.push_back(std::make_pair("Difficulty", RoundToString(GetBlockDifficulty(block.nBits),8)));
+    res.push_back(std::make_pair("CPID", bb.cpid));
+    res.push_back(std::make_pair("Interest", RoundToString(bb.InterestSubsidy,8)));
 
-    res["cpid"]=bb.cpid;
-    res["Magnitude"]=bb.Magnitude;
-    res["Interest"]=bb.InterestSubsidy;
-    res["BoincReward"]=bb.ResearchSubsidy;
-    res["Difficulty"]=GetBlockDifficulty(block.nBits);
+    if (bb.Magnitude > 0)
+    {
+        res.push_back(std::make_pair("Magnitude", RoundToString(bb.Magnitude,8)));
+        res.push_back(std::make_pair("Boinc Reward", RoundToString(bb.ResearchSubsidy,8)));
+    }
 
+    res.push_back(std::make_pair("Is Superblock", (bb.superblock.length() >= 20 ? "true" : "false")));
 
     if(fDebug)
     {
-        res["xbbBoincPublicKey"]=bb.BoincPublicKey;
+        if (bb.superblock.length() >= 20)
+            res.push_back(std::make_pair("Neural Contract Binary Size", ToString(bb.superblock.length())));
```

Comment:
I think this should be near "Neural Hash".

Reply:
its the binary contract. not the hash but i could include the hash in a line as well. maybe we should just reword this

Reply:
I just would like to see superblock and neural related things near each other.

Reply:
--complete